File: src/app/models/Coupon.ts
export class Coupon {
  public constructor(
    public id?: number,
    public companyId?: number,
    public title?: string,
    public startDate?: string,
    public endDate?: string,
    public amount?: number,
    public catagory?: string,
    public description?: string,
    public price?: number,
    public image?: string,
  ) { }
}

export interface CouponType {
  value: string;
}

File: JBCouponProject/src/main/java/com/dansa/JBCouponProject/Beans/Coupon.java
package com.dansa.JBCouponProject.Beans;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Coupon {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	private int companyId;
	@Enumerated(EnumType.STRING)
	private Catagory catagory;
	private String title;
	private String description;
	private Date startDate;
	private Date endDate;
	private int amount;
	private double price;
	private String image;

}
